src/dotnet/Chat.UI.Blazor/Components/ChatList/ChatListItem.razor

```diff
@@ -1,101 +1,147 @@
 @namespace ActualChat.Chat.UI.Blazor.Components
-@inherits ComputedStateComponent<ChatState>
+@inherits ComputedStateComponent<ChatListItem.Model>
 @{
     var m = _rendered = State.Value;
-    var chat = m.Chat;
-    var info = m.Info;
+    if (m == Model.None)
+        return;
+
+    var chatState = m.ChatState;
+    var chat = chatState.Chat;
+    var info = chatState.Info;
     var lastTextEntry = info.LastTextEntry;
-    var audioState = m.AudioState;
+    var audioState = chatState.AudioState;
+    var isLastPinned = m.IsLastPinned;
 }
 
-<NavbarItem
-    Url="@($"/chat/{chat.Id}")"
-    IsSelected="@m.IsSelected"
-    data-menu="@(MenuRef.New<ChatMenu>(chat.Id, ListKind.Format()).ToString())"
-    data-menu-placement="@(FloatingPosition.BottomStart.ToPositionString())">
-    <ChildContent>
-        <div class="flex-1 flex-x items-center gap-x-2" @onclick="()=>OnClick(chat.Id)">
-            <ChatIcon Chat="m.Chat"/>
-            <div class="grid justify-start items-start gap-y-1">
-                <span class="text-headline-1 text-02 truncate">
-                    <SearchUIHighlighter Text="@m.Chat.Title"/>
-                </span>
-                @if (lastTextEntry != null) {
-                    <div class="text-xs text-03 truncate">
-                        @if (m.Chat.Kind != ChatKind.Peer && !lastTextEntry.IsSystemEntry) {
-                            <span>
-                                <AuthorName Class="text-xs text-02 font-medium" AuthorSid="@lastTextEntry.AuthorId" ShowDetailsOnClick="false"/>:
-                            </span>
-                        }
-                        @info.LastTextEntryText
+@if (m == Model.Loading) {
+    <ChatListItemSkeleton />
+}
+else {
+    <NavbarItem
+        Url="@($"/chat/{chat.Id}")"
```

Comment:
why not `Links.Chat(chat.Id)`?

Reply:
Yep.

Reply:
I have copied the code without reading)))

Reply:
Just moved it to another location